packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/indexPatterns/deleteIndex/DeleteIndex.modal.tsx

```diff
@@ -0,0 +1,103 @@
+import { useNavigate, useParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { useEffect } from 'react';
+import { Button } from '@/components/ui/button';
+import {
+  DialogClose,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from '@/components/ui/dialog';
+import { useToast } from '@/components/ui/use-toast';
+import { useDeleteIndex } from '@/hooks/api/database/indexes/useDeleteIndex.hook';
+import { useServiceData } from '../../Service.context';
+import { getCdbApiErrorMessage } from '@/lib/apiHelper';
+import { useGetIndexes } from '@/hooks/api/database/indexes/useGetIndexes.hook';
+import RouteModal from '@/components/route-modal/RouteModal';
+
+const DeleteIndexModal = () => {
+  const { projectId, indexId } = useParams();
+  const navigate = useNavigate();
+  const { service } = useServiceData();
+  const indexesQuery = useGetIndexes(projectId, service.engine, service.id, {
+    enabled: !!service.id,
+  });
+
+  const indexes = indexesQuery.data;
+  const deletedIndex = indexes?.find((i) => i.id === indexId);
+
+  const { t } = useTranslation(
+    'pci-databases-analytics/services/service/indexPatterns',
+  );
+  const toast = useToast();
+  const { deleteIndex, isPending } = useDeleteIndex({
+    onError: (err) => {
+      toast.toast({
+        title: t('deleteIndexToastErrorTitle'),
+        variant: 'destructive',
+        description: getCdbApiErrorMessage(err),
+      });
+    },
+    onSuccess: () => {
+      toast.toast({
+        title: t('deleteIndexToastSuccessTitle'),
+        description: t('deleteIndexToastSuccessDescription', {
+          name: deletedIndex.name,
```

Comment:
As code is executed in a synchronous way before first render, `deletedIndex` could be `undefined`